Sources/Client/Client/Client+Setup.swift

```diff
@@ -52,12 +52,11 @@ extension Client {
         logger?.log("Sending a request for a Guest Token...")
         
         request(endpoint: .guestToken(user)) { [unowned self] (result: Result<TokenResponse, ClientError>) in
-            do {
-                let response = try result.get()
+            if let response = result.value {
                 self.set(user: response.user, token: response.token, completion)
-            } catch {
+            } else if let error = result.error {
```

Comment:
I propose we make it a guideline for us to use `do { } catch { }` with `Result`s instead of `if let`s.

Reply:
I agree, but here `Result` knows it should be a `ClientError`, so no need to cast. From your proposal will be this:
```swift
            do {
                let response = try result.get()
                self.set(user: response.user, token: response.token, completion)
            } catch let error as ClientError {
                self.logger?.log(error, message: "Guest Token")
                completion?(.failure(error))
            } catch {
                // ???
            }
```

Reply:
That makes sense, seems like a swift shortcoming